delira/data_loading/dataset.py

```diff
@@ -407,76 +366,121 @@ def _make_dataset(self, path):
             if `path` is not a valid directory
 
         """
-        data = []
-        assert os.path.isdir(path), '%s is not a valid directory' % dir
-
-        for root, _, fnames in sorted(os.walk(path)):
-            for fname in fnames:
-                fpath = os.path.join(root, fname)
-                if self._is_valid_image_file(fpath):
-                    sample = [fpath]
-                    for ext in self._gt_extensions:
-                        gt_path = fpath.rsplit(".", maxsplit=1)[0] + ext
-                        if os.path.isfile(gt_path):
-                            sample.append(gt_path)
-
-                    data.append(self._load_fn(
-                        *sample, **self._load_kwargs))
-
+        if isinstance(path, list):
+            # generate list from iterable
+            data = list(path)
+        else:
+            # generate list from all items
+            assert os.path.isdir(path), '%s is not a valid directory' % dir
+            data = [os.path.join(path, p) for p in os.listdir(path)]
         return data
 
-    def _is_valid_image_file(self, fname):
+    def __getitem__(self, index):
         """
-        Helper Function to check wheter file is image file and has at least
-        one label file
+        load data sample specified by index
 
         Parameters
         ----------
-        fname : str
-            filename of image path
+        index : int
+            index to specifiy which data sample to load
 
         Returns
         -------
-        bool
-            is valid data sample
+        dict
+            loaded data sample
+        """
+        data_dict = self._load_fn(self.get_sample_from_index(index),
+                                  **self._load_kwargs)
+        return data_dict
+
+
+class BaseExtendCacheDataset(BaseCacheDataset):
```

Comment:
I quite don't get the difference of `BaseCacheDataset` and `BaseExtendCacheDataset`. From the docstring they should be the same?

Reply:
The `BaseCacheDataset` uses the `append` to add elements to the internal data list. `BaseExtendCacheDataset` uses `extend`, so the load_fn is expected to return an iterator which is added to the internal data list -> this should enables loading of multiple samples at once
I'm going to change the doc string so it is more clear to everybody.